### DISCUSSION TITLE
how to do something after the first load success

### DISCUSSION THREAD

**Discussion by KingAmo** [TanStack/query#1245] [Q&A]
i want to do something only when the first time load success,  how to do that please?

**Comment by TkDodo** [comment]
Can you be more specific, what do you want todo, and why only for the first time? Should the query still re-fetch? If not - staleTime: Infinity and onSuccess callback could work. Otherwise, have a useEffect and listen to the isLoading flag maybe? But it’s not guaranteed to run only once
